tough/src/key_source.rs

```diff
@@ -17,11 +17,38 @@ pub trait KeySource: Debug + Send + Sync {
     fn as_sign(&self) -> Result<Box<dyn Sign>, Box<dyn std::error::Error + Send + Sync + 'static>>;
 
     /// Writes a key back to the `KeySource`
-    fn write(
+    fn write(&self, value: &str) -> Result<(), Box<dyn std::error::Error + Send + Sync + 'static>>;
+    /// Creates a key for key source, default implementation creates rsa key pair which can
+    /// be override by individual `KeySource` implementation
+    fn create(
```

Comment:
Hmmm, I'm a little worried about a "default" implementation for this method that assumes `openssl` is installed.  

I'm also wondering why we need a `create` method?  Is it a convenience just for `tuftool`?

Reply:
I'm also wondering why we need a create method? Is it a convenience just for tuftool?
Indirectly Yes, create is used by gen-rsa-key which is for convenience.
But, it helps us with end to end test case and programmatic key generation wherever required.